### ISSUE TITLE
Incorrect test split start point and length when using frequency with unit >1

### ISSUE THREAD

**Issue by gorold** [awslabs/gluon-ts#2499]
## Description
(A clear and concise description of what the bug is.)

The behaviour of ```split``` is not as expected when using a dataset and split point with frequency that has larger unit than 1, e.g. 30T, 6M, etc. This causes the training set length and test set lengths to be incorrect.

This likely stems from the ```to_integer_slice``` function not behaving as expected when sliced with ```pd.Period``` with frequency that is larger than 1 unit, e.g. 30T, 6M, etc. Related issue: #1833 but code seems to have changed significantly since.

Issue seems to be due to this. Could consider doing something like ```start_offset = (slice_.start - start).n // pd.tseries.frequencies.to_offset(start.freq).n```.
https://github.com/awslabs/gluonts/blob/cac5e6b91cbd545c0811b20849290e9215c1b822/src/gluonts/dataset/split.py#L111
https://github.com/awslabs/gluonts/blob/cac5e6b91cbd545c0811b20849290e9215c1b822/src/gluonts/dataset/split.py#L122

## To Reproduce
(Please provide minimal example of code snippet that reproduces the error. For existing examples, please provide link.)

```python
import pandas as pd
import numpy as np
from gluonts.dataset.pandas import PandasDataset
from gluonts.dataset.split import split

n = 100
split_n = 20
freq = '30T'
date = "2015-04-07 00:00:00"

df = pd.DataFrame(np.random.randn(n), index=pd.period_range(date, periods=n, freq=freq))
dataset = PandasDataset(df, target=0)
split_date = pd.Period(date, freq=freq) + pd.to_timedelta(freq) * (split_n - 1)
training_dataset, test_template = split(
    dataset, date=split_date
)
print(df.index[0])
print(split_date)
print(df.index[-1])
print(next(iter(training_dataset))['target'].shape)
print(next(iter(test_template.generate_instances(prediction_length=6)))[0]['target'].shape)
print(next(iter(test_template.generate_instances(prediction_length=6)))[1]['target'].shape)
```

## Error message or code output
(Paste the complete error message, including stack trace, or the undesired output that the above snippet produces.)

```
2015-04-07 00:00
2015-04-07 09:30
2015-04-09 01:30
(100,)
(100,)
(0,)
```

## Expected message
```
2015-04-07 00:00
2015-04-07 09:30
2015-04-09 01:30
(20,)
(20,)
(6,)
```

## Environment
- GluonTS version: 0.11.4

(Add as much information about your environment as possible, e.g. dependencies versions.)

**Comment by lostella** [comment]
Thanks @gorold for the great bug report! A fix is in the making in #2500, I just want to make sure everything is tested properly
